pkg/streamingpromql/engine_test.go

```diff
@@ -241,6 +240,28 @@ func TestNewRangeQuery_InvalidExpressionTypes(t *testing.T) {
 	require.EqualError(t, err, "query expression produces a string, but expression for range queries must produce an instant vector or scalar")
 }
 
+func TestNewInstantQuery_Strings(t *testing.T) {
+	ctx := context.Background()
+	opts := NewTestEngineOpts()
+	mimirEngine, err := NewEngine(opts, NewStaticQueryLimitsProvider(0), stats.NewQueryMetrics(nil), log.NewNopLogger())
+	require.NoError(t, err)
+
+	prometheusEngine := promql.NewEngine(opts.CommonOpts)
+
+	storage := promqltest.LoadedStorage(t, ``)
```

Comment:
It should be possible to drop this and pass `nil` to `NewInstantQuery` instead - the storage won't be used because there are no selectors.

Reply:
This is true for MQE, but Prometheus panic's without it. (And since this isn't a real scenario, I didn't dig deeper and just passed the storage into both engines).

Reply:
Weird, alright, let's leave it then.